File: DataBase/tfg/src/main/java/com/tfg_data_base/tfg/GameInfo/GameInfo.java
package com.tfg_data_base.tfg.GameInfo;

import java.util.ArrayList;
import java.util.List;
import java.util.Map;

import org.springframework.data.mongodb.core.mapping.Document;

import lombok.Data;

@Document(value = "GameInfo")
@Data
public class GameInfo {
    // El id sera unico ya que solo exitira un json con la informacion
    private String id = "GAME_INFO_ID";
    private List<Critteron> critterons = new ArrayList<>();
    private List<Furniture> furniture = new ArrayList<>();
    private WeekRewards weekRewards;

    @Data
    public static class Critteron {
        private String critteronID;
    }

    @Data
    public static class Furniture {
        private String furnitureID;
    }

    @Data
    public static class WeekRewards {
        private Map<String, DayReward> days;

        @Data
        public static class DayReward {
            private Integer reward1;
            private Integer reward2;

            public DayReward(Integer reward1, Integer reward2) {
                this.reward1 = reward1;
                this.reward2 = reward2;
            }
        }
    }
}

File: TFG-Critterons-App/Assets/Test/Scripts/GameInfo.cs
using System;
using System.Collections.Generic;

[Serializable]
public class GameInfo   {
    public string id;
    public List<Critteron> critterons;
    public List<Forniture> forniture;
    public List<User> users;
    public WeekRewards weekRewards;

    [Serializable]
    public class Critteron  {
        public string critteronID;
    }

    [Serializable]
    public class Forniture  {
        public string fornitureID;
    }

    [Serializable]
    public class User   {
        public string userID;
    }

    [Serializable]
    public class WeekRewards     {
        public Dictionary<string, DayReward> days;

        [Serializable]
        public class DayReward  {
            public int reward1;
            public int reward2;
        }
    }
}